### PR TITLE
Pipeline configuration And scheme modules

### PR DESCRIPTION
**Pull request by IsaackMwenda**

### CODE REVIEW

File: project_test/lib/test_schemes.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import json
+
+from core_data_modules.data_models import Scheme
+
+def _open_scheme(filename):
+    with open(f"code_schemes/{filename}","r") as f:
```
Comment (as2388) [inline project_test/lib/test_schemes.py]: There should be a space after the comma here

File: project_test/lib/pipeline_configurations.py [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+            id_field = "{}_id".format(self.raw_field)
+        self.run_id_field = id_field
+
+class PipelineConfiguration(object):
+    
+    DEV_MODE = False
+
+    TEST_SHOWS_CODING_PLAN = [
+        CodingPlan(raw_field="internet_working_raw",
+        coded_field="internet_working_coded",
```
Comment (as2388) [inline project_test/lib/pipeline_configurations.py]: Indentation

File: project_test/lib/pipeline_configurations.py [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+        self.run_id_field = time_field
+        self.analysis_file_key = analysis_file_key
+
+        if id_field is None:
+            id_field = "{}_id".format(self.raw_field)
+        self.run_id_field = id_field
+
+class PipelineConfiguration(object):
+    
+    DEV_MODE = False
```
Comment (as2388) [inline project_test/lib/pipeline_configurations.py]: You probably want this to be True for now.

Apologies for there not being a comment saying what it's for - when True, data from test users will be included in the analysis file, which is really useful when you're building pipelines. 
Also, this should probably have been a command line flag...

File: project_test/lib/pipeline_configurations.py [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+        coda_filename="water_filter.json"
+        icr_filename="water_filter_icr.csv",
+        run_id_field="water_filter_id",
+        analysis_file_key="water_filter_",
+        cleaner=None,
+        code_scheme=CodeSchemes.WATER),
+    ]
+    
+    @staticmethod
+    def test_clean_age(text):
```
Comment (as2388) [inline project_test/lib/pipeline_configurations.py]: Maybe call this `clean_age_within_range` or similar.

`test_clean_age` sounds too much like a unittest method.

File: project_test/lib/pipeline_configurations.py [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+        CodingPlan(raw_field="age_raw",
+        coded_field="age_coded",
+        time_field="age_time",
+        coda_filename="age.json",
+        analysis_file_key="age",
+        cleaner=lambda text: PipelineConfiguration.test_clean_age(text),
+        code_scheme=CodeScheme.AGE),
+    ]
+
+    SURVEY_CODING_PLANS.extend([
```
Comment (as2388) [inline project_test/lib/pipeline_configurations.py]: Why not include this in the previous list?

File: project_test/lib/pipeline_configurations.py [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+        coded_field="gender_coded",
+        time_field="gender_time",
+        coda_filename="gender.json"
+        run_id_field="gender_id",
+        analysis_file_key="gender",
+        cleaner=swahili.DemographicCleaner.clean_gender,
+        code_scheme=CodeSchemes.GENDER)
+    ])
+        
+    LOCATION_CODING_PLANS = [
```
Comment (as2388) [inline project_test/lib/pipeline_configurations.py]: Ditto

**Review by as2388 — APPROVED** [review]
Approved with comments

**Comment by IsaackMwenda** [comment]
Thanks @as2388 . @lukechurch PTAL.

**Review by lukechurch — APPROVED** [review]

**Pull request merged**